packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/TermsOfServicesStep/TermsOfServicesStep.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useRef } from 'react';
+import { Link } from 'react-router-dom';
+
+import { CWCheckbox } from 'views/components/component_kit/cw_checkbox';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import {
+  CWForm,
+  CWFormRef,
+} from 'views/components/component_kit/new_designs/CWForm';
+import { termsOfServicesFormValidation } from './validations';
+
+import './TermsOfServicesStep.scss';
+
+type TermsOfServicesStepProps = {
+  onComplete: () => void;
+};
+
+const TermsOfServicesStep = ({ onComplete }: TermsOfServicesStepProps) => {
+  const formMethodsRef = useRef<CWFormRef>();
+
+  const handleSubmit = () => {
+    onComplete();
+  };
+
+  return (
+    <CWForm
+      // @ts-expect-error <StrictNullChecks/>
+      ref={formMethodsRef}
+      className="TermsOfServicesStep"
+      initialValues={{
+        enableTermsOfServices: true,
+      }}
+      validationSchema={termsOfServicesFormValidation}
+      onSubmit={handleSubmit}
+    >
+      {({ watch }) => {
+        const isTermsChecked = watch('enableTermsOfServices');
+        return (
+          <React.Fragment>
+            <iframe
+              className="terms-frame"
+              allowFullScreen={true}
+              src={`${window.location.origin}/terms`}
+              loading="eager"
+            />
```

Comment:
cant we just use <TermsPage /> component (with some css adjustments if necessary) instead of iframe?

Reply:
this is requirement in ticket by dillion  @masvelio

Reply:
I know it says to use an iframe, but I agree with @masvelio to try to use a component with any required css adjustments instead. Also is there any way to get the Mava widget to not show on the ToS screen?

Reply:
@Israellund  okay i will try to use /terms component or i will  design one